### PR TITLE
PS-3840: With FIPS enabled and performance_schema=off, instance crash…

### PR DESCRIPTION
**Pull request by satya-bodapati**
…es on CREATE VIEW

Description:
-----------
MySQL Server compiled with OpenSSL (not YaSSL) and running the
server under FIPS enabled environment, leads to crash.

This is because MD5 algorithm is not FIPS compliant.

Fix:
----
Disable FIPS mode for MySQL which allows mysqld to be run under
FIPS environment.

### CODE REVIEW

File: sql/mysqld.cc
Comment (laurynas-biveinis) [inline sql/mysqld.cc]: While the statement is true for MySQL too in general, the product here is Percona Server

File: sql/mysqld.cc
Comment (laurynas-biveinis) [inline sql/mysqld.cc]: OpenSSL docs suggest plain "int" as FIPS_mode return type

File: extra/fipstest.cc
Comment (laurynas-biveinis) [inline extra/fipstest.cc]: don't

File: mysql-test/include/have_fips.inc
Comment (laurynas-biveinis) [inline mysql-test/include/have_fips.inc]: While the utility app works everywhere, I wonder if a shell-based test would be enough for all of our practical testing use cases - i.e. on CentOS, `cat /proc/sys/crypto/fips_enabled`. We probably won't be testing FIPS on Debian/Ubuntu anyway for time being

File: mysql-test/t/fips.test
Comment (laurynas-biveinis) [inline mysql-test/t/fips.test]: grep mysqld.1.err is an antipattern. It breaks --repeat (the 2nd invocation will just hit the 1st one's output), it will break if any of the previous tests on the same worker got the same warning. Deleting the log file to fix the above is not good neither. The best solution I know so far is to restart the server with custom error log path, grep, restart with default, delete the custom error log.

File: extra/fipstest.cc
Comment (laurynas-biveinis) [inline extra/fipstest.cc]: Just printf I guess

File: extra/fipstest.cc
Comment (laurynas-biveinis) [inline extra/fipstest.cc]: (see the have_fips.inc review comments first)

File: extra/fipstest.cc
Comment (laurynas-biveinis) [inline extra/fipstest.cc]: printf

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: sql/mysqld.cc
Comment (satya-bodapati) [reply sql/mysqld.cc]: Done! changed to Percona Server

File: extra/fipstest.cc
Comment (satya-bodapati) [reply extra/fipstest.cc]: removed this file

File: mysql-test/include/have_fips.inc
Comment (satya-bodapati) [reply mysql-test/include/have_fips.inc]: Ok, my approach was more generic. I have implemented the suggestion. The change looks smaller now :)

File: mysql-test/t/fips.test
Comment (satya-bodapati) [reply mysql-test/t/fips.test]: Fixed. the test is now named as percona_fips.test

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
The diff itself looks good, other comments:
- please squash commits
- please provide Jenkins run link (in fact I prefer requesting PRs for my work when Jenkins completes and I review it, others have other styles)

**Comment by satya-bodapati** [comment]
Jenkins: https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1842/

**Comment by laurynas-biveinis** [comment]
Jenkins checks out OK. Before merging:
- please squash your commits
- please do not merge from trunk (rebase on the trunk instead but in this case there is no pressing need unless you get merge conflicts otherwise)

**Pull request merged**
